go-controller/pkg/clustermanager/network_cluster_controller.go

```diff
@@ -62,6 +63,9 @@ type networkClusterController struct {
 	subnetAllocator     subnet.Allocator
 
 	util.NetInfo
+
+	// TODO: this should be a watch grabbed from the watch factory
+	nadClient networkattchmentdefclientset.Interface
```

Comment:
this is a leftover; but we'll need the NAD watcher to plumb into the allocator, so we can figure out what's the namespace's primary network

Reply:
i did this already in my PR there is NAD watcher